#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023a/projects/objects/floors/protos/Floor.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/Rubber.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/tables/protos/Table.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/kitchen/utensils/protos/Plate.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/kitchen/breakfast/protos/BiscuitBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/kitchen/breakfast/protos/JamJar.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/kitchen/utensils/protos/Glass.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/appearances/protos/RoughPine.proto"

WorldInfo {
  basicTimeStep 16
  contactProperties [
    ContactProperties {
      material2 "free_wheel"
      coulombFriction [
        0
      ]
    }
  ]
}
Viewpoint {
  orientation -0.1796852792162284 -0.05998338135324404 0.9818936777443968 3.7748552421575043
  position 1.5611038927753451 1.1873265218561329 1.0594483967621502
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 5 5
}
Robot {
  translation 0 0 0.1
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0.125 0
      }
      device [
        PositionSensor {
          name "right wheel sensor"
        }
        RotationalMotor {
          name "right wheel motor"
        }
      ]
      endPoint Solid {
        translation 0 0.124999 0
        rotation 1 -2.551395949956219e-16 -3.1401796307153465e-16 1.5707996938995055
        children [
          DEF WHEEL_SHAPE Shape {
            appearance Rubber {
            }
            geometry Cylinder {
              height 0.05
              radius 0.1
            }
          }
        ]
        name "Right Wheel"
        boundingObject USE WHEEL_SHAPE
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 -0.125 0
      }
      device [
        PositionSensor {
          name "left wheel sensor"
        }
        RotationalMotor {
          name "left wheel motor"
        }
      ]
      endPoint Solid {
        translation 0 -0.125 0
        rotation 1 0 3.1401934450118313e-16 1.5707908955163137
        children [
          USE WHEEL_SHAPE
        ]
        name "Left Wheel"
        boundingObject USE WHEEL_SHAPE
        physics Physics {
        }
      }
    }
    Solid {
      children [
        DEF BODY_SHAPE Shape {
          appearance Appearance {
            material Material {
              diffuseColor 0 0.333333 1
            }
          }
          geometry Box {
            size 0.1 0.25 0.1
          }
        }
      ]
      name "Body"
      boundingObject USE BODY_SHAPE
      physics Physics {
      }
    }
    Accelerometer {
    }
    InertialUnit {
    }
    GPS {
    }
    Gyro {
    }
    Group {
      children [
        SliderJoint {
          jointParameters JointParameters {
            position 0.15
            maxStop 0.3
            springConstant 10
            dampingConstant 5
            staticFriction 100
          }
          device [
            LinearMotor {
              maxPosition 0.3
            }
          ]
          endPoint Solid {
            translation 0 0 0.35000000000000053
            children [
              DEF UPPER_STICK_SHAPE Group {
                children [
                  Shape {
                    appearance PBRAppearance {
                      baseColor 0.521569 0.521569 0.521569
                      metalness 0
                    }
                    geometry Cylinder {
                      height 0.3
                      radius 0.01
                    }
                  }
                  Pose {
                    translation 0 0 0.155
                    children [
                      Shape {
                        appearance PBRAppearance {
                          baseColor 0.521569 0.521569 0.521569
                          metalness 0
                        }
                        geometry Box {
                          size 0.3 0.3 0.01
                        }
                      }
                    ]
                  }
                ]
              }
            ]
            boundingObject USE UPPER_STICK_SHAPE
            physics Physics {
            }
          }
        }
        Solid {
          translation 0 0 0.2
          rotation 1 0 0 0
          children [
            DEF STICK_SHAPE Shape {
              appearance PBRAppearance {
                baseColor 0.521569 0.521569 0.521569
                metalness 0
              }
              geometry Cylinder {
                height 0.3
                radius 0.01
              }
            }
          ]
          name "stick"
          boundingObject USE STICK_SHAPE
          physics Physics {
          }
        }
      ]
    }
    DistanceSensor {
      translation 0.055 0.041 0
      children [
        DEF DIS_SENSOR_SHAPE Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "FL distance sensor"
    }
    DistanceSensor {
      translation 0.055 -0.041 0
      children [
        DEF DIS_SENSOR_SHAPE Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "FR distance sensor"
    }
    DistanceSensor {
      translation -0.055 0.041 0
      rotation 0 0 1 3.141592653589793
      children [
        DEF DIS_SENSOR_SHAPE Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "RL distance sensor"
    }
    DistanceSensor {
      translation -0.055 -0.041 0
      rotation 0 0 1 3.141592653589793
      children [
        DEF DIS_SENSOR_SHAPE Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "RR distance sensor"
    }
  ]
  boundingObject USE BODY_SHAPE
  physics Physics {
  }
  controller "inverted_pendulum"
}
Table {
  translation -2.03446 -1.54056 1.08358e-13
  rotation 0 0 1 1.5707996938995747
  size 1.5 0.9 0.61
  feetSize 0.05 0.05
}
Plate {
  translation -1.95229 -1.37933 0.67
}
JamJar {
  translation -2.1053 -1.22437 0.673
}
Glass {
  translation -1.96661 -1.22849 0.672
}
BiscuitBox {
  translation -2.11564 -1.41803 0.672
}
Solid {
  translation -2.01496 -1.36277 0.665
  rotation 0 0 1 1.5707996938995747
  children [
    DEF TRAY_GROUP Group {
      children [
        Pose {
          translation 0 -0.15 -0.03
          children [
            DEF TRAY_BASE_SHAPE Shape {
              appearance RoughPine {
              }
              geometry Box {
                size 0.4 0.01 0.05
              }
            }
          ]
        }
        Pose {
          translation 0 0.15 -0.03
          children [
            USE TRAY_BASE_SHAPE
          ]
        }
        Pose {
          children [
            Shape {
              appearance RoughPine {
              }
              geometry Box {
                size 0.4 0.35 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  boundingObject USE TRAY_GROUP
  physics Physics {
  }
}
Robot {
  translation -2.00158 -1.84608 0.635
  children [
    Solid {
      translation -0.08 0 -0.015
      children [
        DEF AUX_ROBOT_FREE_WHEEL_SHAPE Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Sphere {
            radius 0.01
          }
        }
      ]
      name "solid(4)"
      contactMaterial "free_wheel"
      boundingObject USE AUX_ROBOT_FREE_WHEEL_SHAPE
      physics Physics {
      }
    }
    Solid {
      translation 0.08 0 -0.015
      children [
        USE AUX_ROBOT_FREE_WHEEL_SHAPE
      ]
      name "solid(5)"
      contactMaterial "free_wheel"
      boundingObject USE AUX_ROBOT_FREE_WHEEL_SHAPE
      physics Physics {
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0 -0.01
      }
      device [
        RotationalMotor {
          name "robot1 left wheel motor"
        }
      ]
      endPoint Solid {
        translation 0 -0.08 -0.01
        rotation 1 0 0 1.5707996938995747
        children [
          DEF AUX_ROBOT_WHEEL_SHAPE Shape {
            appearance PBRAppearance {
              baseColor 0.756863 0.756863 0.756863
              metalness 0
            }
            geometry Cylinder {
              height 0.01
              radius 0.015
            }
          }
        ]
        name "solid(1)"
        boundingObject USE AUX_ROBOT_WHEEL_SHAPE
        physics Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0 -0.01
      }
      device [
        RotationalMotor {
          name "robot1 right wheel motor"
        }
      ]
      endPoint Solid {
        translation 0 0.08 -0.01
        rotation 1 0 0 1.5707996938995747
        children [
          USE AUX_ROBOT_WHEEL_SHAPE
        ]
        name "solid(3)"
        boundingObject USE AUX_ROBOT_WHEEL_SHAPE
        physics Physics {
        }
      }
    }
    Pose {
      translation 0 0 0.001
      children [
        SliderJoint {
          jointParameters JointParameters {
            minStop -0.001
            maxStop 0.015
            staticFriction 50
          }
          device [
            LinearMotor {
              name "robot1 linear motor"
              maxPosition 0.015
            }
          ]
          endPoint Solid {
            translation 0 1.9317880628477724e-14 0.017000000000000015
            rotation -0.9739004641262347 -0.22697551812734532 1.206350480078731e-05 1.4072491669460728e-16
            children [
              Pose {
                translation 0 0 -0.01
                children [
                  Shape {
                    appearance PBRAppearance {
                      baseColor 0.792157 0.792157 0.792157
                      metalness 0
                    }
                    geometry Cylinder {
                      height 0.02
                      radius 0.07
                    }
                  }
                ]
              }
              DEF AUX_ROBOT_PLATE Shape {
                appearance PBRAppearance {
                  baseColor 0.792157 0.792157 0.792157
                  metalness 0
                }
                geometry Cylinder {
                  height 0.005
                  radius 0.1
                }
              }
            ]
            name "solid(2)"
            boundingObject USE AUX_ROBOT_PLATE
            physics Physics {
            }
          }
        }
      ]
    }
    Solid {
      children [
        DEF AUX_ROBOT_BODY Shape {
          appearance PBRAppearance {
            baseColor 0.305882 0.305882 0.305882
            metalness 0
          }
          geometry Cylinder {
            height 0.03
            radius 0.1
          }
        }
      ]
    }
  ]
  name "robot(1)"
  boundingObject USE AUX_ROBOT_BODY
  physics Physics {
  }
  controller "aux_robot"
}
